src/lib/eventHandler/pasteHandler.ts

```diff
@@ -0,0 +1,160 @@
+import { isImageResponse } from '../http'
+import { IpcRenderer } from 'electron'
+import isElectron from 'is-electron'
+
+let ipcRenderer: IpcRenderer | null = null
+if (window.require !== undefined) {
+  // avoid for 'fs.existsSync is not a function'
+  // see https://github.com/electron/electron/issues/7300#issuecomment-248773783
+  ipcRenderer = window.require('electron').ipcRenderer
+}
+
+interface PasteInfo {
+  doc: CodeMirror.Doc
+  change: CodeMirror.EditorChange
+  pastedText: string
+}
+
+interface FetchResult {
+  contentType: string
+  body: string
+  isSuccess: boolean
+}
+
+let info: PasteInfo
+
+const handlePasteText = (
+  doc: CodeMirror.Doc,
+  change: CodeMirror.EditorChange,
+  pastedText: string
+) => {
+  doc.replaceRange(pastedText, change.from, {
+    line: change.from.line,
+    ch: change.from.ch + pastedText.length,
+  })
+}
+
+const handlePasteUrl = async (doc: CodeMirror.Doc, change: CodeMirror.EditorChange, pastedText: string) => {
+  if (ipcRenderer === null) return
+  info = { doc, change, pastedText }
+
+  const taggedUrl = `<${pastedText}>`
+  const urlToFetch = pastedText
+
+  doc.replaceRange(taggedUrl, change.from, {
+    line: change.from.line,
+    ch: change.from.ch + pastedText.length,
+  })
+
+  ipcRenderer.send('fetch-page-title-request', urlToFetch)
+}
+
+if (ipcRenderer !== null) {
+  ipcRenderer.on('fetch-page-title-response', async (_, fetchResult: FetchResult) => {
+    let replacement = info.pastedText
+    if (!isImageResponse(fetchResult.contentType)) {
```

Comment:
Instead of checking if the response is a image response. Why don't we check for `text/html` content type only? This will cover other cases for other files such as video and audio.